java/memory/src/main/java/org/apache/arrow/memory/UnsafeAllocationManager.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.memory;
+
+import org.apache.arrow.memory.util.MemoryUtil;
+
+/**
+ * Allocation manager based on unsafe API.
+ */
+public class UnsafeAllocationManager extends AllocationManager {
+
+  public static final Factory FACTORY = new Factory();
+
+  private final long allocatedSize;
+
+  private final long allocatedAddress;
+
+  protected UnsafeAllocationManager(BaseAllocator accountingAllocator, long requestedSize) {
```

Comment:
should this be private? Perhaps the class should be final too?

Reply:
Sounds good. 
I have made it package private (to make testing easier), and made the class final.
Please check it. Thank you.